### ISSUE TITLE
Add metrics for chain fork work and lengths

### ISSUE THREAD

**Issue by teor2345** [ZcashFoundation/zebra#5297]
## Motivation

The ZIP editors asked for the following Zebra metrics:
- histogram of chain work for all recent chain forks
- histogram of tip heights for all recent chain forks
- histogram of non-finalized lengths for all recent chain forks
- histogram of fork heights for all recent chain forks
- histogram of fork lengths (`tip height - fork height`)
- all the above metrics as a gauge for the best chain only

In #6235, we added APIs for these metrics, but we need to track them using prometheus in this ticket:
- fork heights
- fork lengths

We already implement these metrics
- number of recent forks
- some of the above metrics (for the best chain only)

`zcashd` is currently unable to track these metrics, because it doesn't seek out multiple chains in parallel.

**Comment by mpguerra** [comment]
When would they like these by? The earliest I would schedule these would be next sprint (21) and only if the release candidate work is complete

**Comment by daira** [comment]
This is for ZIP 315, to verify whether the recommended "trusted" and "untrusted" confirmation depths make sense and to provide supporting rationale. So not immediately urgent.

**Comment by teor2345** [comment]
This is blocked by PR #5257 merging, because it changes some of the same code as this ticket.

**Comment by mpguerra** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Engineering-Team-5fb24d9264a3e8000e666a9e/issues/ZcashFoundation/zebra/5297?planning-poker) with Zenhub @arya2 @conradoplg @oxarbitrage @teor2345 @upbqdn

**Comment by mpguerra** [comment]
tentatively scheduling this for after the GBT work

**Comment by teor2345** [comment]
Some of these changes are already made in PR #6235, when I get time, I'll split them out into a separate PR.

**Comment by teor2345** [comment]
> Some of these changes are already made in PR #6235, when I get time, I'll split them out into a separate PR.

I updated this ticket with the new metrics APIs that are part of #6235, and the extra things we need to do in this ticket.

**Comment by teor2345** [comment]
We probably want to delay this until after we've renamed some parts of the state in response to an audit finding.

**Comment by teor2345** [comment]
This might also depend on #7109, because at the moment we're storing the partial work of the fork, not the cumulative work of the entire chain.
